core/src/impl/Kokkos_Error.hpp

```diff
@@ -97,7 +97,11 @@ namespace Impl {
 #define KOKKOS_IMPL_ABORT_NORETURN
 #endif
 
-#ifdef KOKKOS_ENABLE_SYCL  // FIXME_SYCL
+// FIXME_SYCL
+// Accomodate host pass for device functions that are not [[noreturn]]
+#if defined(KOKKOS_ENABLE_SYCL) ||  \
+    (defined(KOKKOS_ENABLE_CUDA) && \
+     (defined(__APPLE__) || defined(KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK)))
```

Comment:
How did you come up with this?  I thought CUDA is not supported on OSX.

Reply:
See https://github.com/kokkos/kokkos/blob/d0f710de8010a476abb93f9f63a0b0bc27f7541f/core/src/impl/Kokkos_Error.hpp#L72-L80 and https://github.com/kokkos/kokkos/blob/d0f710de8010a476abb93f9f63a0b0bc27f7541f/core/src/Cuda/Kokkos_Cuda_abort.hpp#L67-L94.